playbook/app/pb_kits/playbook/pb_timestamp/_timestamp.jsx

```diff
@@ -2,36 +2,116 @@
 
 import React from 'react'
 import classnames from 'classnames'
-
+import DateTime from '../pb_kit/dateTime.js'
+import { buildAriaProps, buildCss, buildDataProps } from '../utilities/props'
 import { Caption } from '../'
-
-import { buildCss, buildDataProps } from '../utilities/props'
-
 import { globalProps } from '../utilities/globalProps.js'
 
 type TimestampProps = {
+  align?: "left" | "center" | "right",
+  aria?: object,
+  className?: string | array<string>,
+  dark?: boolean,
+  data?: string,
+  text: string,
+  timestamp: string,
+  timezone: string,
   id?: string,
-  data?: object,
-  className?: string,
-  text?: string,
+  showDate?: boolean,
+  showUser?: boolean,
+  showTimezone?: boolean,
+  variant?: "default" | "elapsed" | "updated"
 }
 
 const Timestamp = (props: TimestampProps) => {
-  const { id, className, data = {}, text } = props
+  const {
+    align = 'left',
+    aria = {},
+    className,
+    dark = false,
+    data = {},
+    text,
+    timestamp,
+    timezone,
+    showDate = true,
+    showUser = false,
+    showTimezone = false,
+    variant = 'default',
+  } = props
+
+  const ariaProps = buildAriaProps(aria)
   const dataProps = buildDataProps(data)
-  const pbCss = buildCss('pb_timestamp_kit')
+  const classes = classnames(
+    buildCss('pb_timestamp_kit', align, {
+      dark: dark,
+      variant: variant,
+    }),
+    globalProps(props),
+    className
+  )
+
+  const currentYear = new Date().getFullYear().toString()
+  const dateTimestamp = new DateTime({ value: timestamp, zone: timezone })
+  const dateDisplay = `${dateTimestamp.toMonth()} ${dateTimestamp.toDay()}`
+  const timeDisplay = `${dateTimestamp.toHour()}:${dateTimestamp.toMinute()}${dateTimestamp.toMeridian()}`
+
+  const fullTimeDisplay = (dateTimestamp, timeDisplay, timezone, showTimezone) => {
+    if (showTimezone == true && timezone.length > 0) {
+      timeDisplay = timeDisplay + ' ' + dateTimestamp.toTimezone()
+    }
+    return timeDisplay
+  }
+
+  const fullDateDisplay = (dateTimestamp, currentYear, timezone, showTimezone) => {
+    let fullDisplay = `${dateTimestamp.toMonth()} ${dateTimestamp.toDay()}`
+    if (dateTimestamp.toYear() > currentYear) {
+      fullDisplay = `${fullDisplay}, ${dateTimestamp.toYear()}`
+    }
+    return `${fullDisplay} ${' \u00b7 '} ${fullTimeDisplay(dateTimestamp, timeDisplay, timezone, showTimezone)}`
+  }
+
+  const formattedUpdatedString = (showUser, text, dateTimestamp, timeDisplay) => {
+    switch (variant) {
+    case 'elapsed': {
+      const elapsedUserDisplay = (showUser == true && text.length > 0) ? ` by ${text}` : ''
```

Comment:
Similar to the comment above in the template, it makes sense to break this down a little more for readability here. 

```suggestion
      const shouldShowUser = showUser == true && text.length > 0
      const elapsedUserDisplay = shouldShowUser ? ` by ${text}` : ''
```

Now it will be easier to re-visit this area and edit the string interpolation area without having to visually parse through the ternary logic comparator as well.